drafts/rxjs-memory-leaks.md

```diff
@@ -0,0 +1,223 @@
+
+### Memory Leaks in a Nutshell
+
+In Angular web applications, memory leaks are caused by a mismanagement of [RxJs Observables](https://rxjs.dev/guide/observable). Thereby, allocated heap memory space is not freed up by the application when a component with an Observable is destroyed. This memory leak will result in performance issues, i.e. the page gets slower and slower. Higher load times negatively impact the user experience and increase the bounce rate [[1]](https://www.nngroup.com/articles/response-times-3-important-limits/). If a page load goes from 1s to 5s, then the bounce rate increases by 90% [[2]](https://www.thinkwithgoogle.com/consumer-insights/consumer-trends/mobile-page-speed-new-industry-benchmarks/). A page refresh will clear the heap memory again, and "reset" the memory leak. So, the longer an Angular component lives on your page, the higher the chance of experiencing a memory leak.
+
+Luckily, we can easily prevent those memory leaks. But first, let us go into more detail about the origins of memory leaks.
+
+### RxJs Observables
+
+Let us start at the beginning - what are [RxJs Observables](https://rxjs.dev/guide/observable)? The _Reactive Extensions for Javascript_ (RxJs), is a library for reactive programming concerned with asynchronous data streams and the propagation of data changes. Thereby, you can consider a data stream as a _collection_ of data arriving and potentially changing over time, e.g. when fetching data from backend services using HTTP.
+
+To that end, RxJS provides one core type - the _Observable_ - representing a collection of future values and events. Observables are lazy push-based systems, similar to JavaScript [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global%5FObjects/Promise). The consumer of the data is unaware of when it will receive the data. It is the data producer that determines when to "push" data to its consumers. Compared to Promises, however, an Observable can produce multiple values over time, i.e. a stream of data. This data stream can return zero to potentially infinite values from its invocation onward, synchronously or asynchronously.
+
+![RxJs Observables](https://cdn.hashnode.com/res/hashnode/image/upload/v1681311720510/6W-qbqxuM.png?auto=compress align="center")
+
+Observables are lazy computations, so, unless you _subscribe_ to them, no data will be transmitted. In Angular applications, we use Observables mostly to fetch data from backend services using HTTP, for routing purposes, and to respond to user events e.g. in forms.
+
+**Please note**: There are Observables that auto-complete, i.e. don't need a specific unsubscribe. This is the case for Angular HttpClient or Angular Router Observables [[3]](https://lukaonik.medium.com/do-we-need-to-unsubscribe-http-client-in-angular-86d781522b99). However, we still recommend applying the same mechanisms to prevent memory leaks as with other Observables, ensuring consistency among the code.
```

Comment:
Broader, you could distinguish between finite and infinite observables. Only the latter one need to be unsubscribed. RxJS can take care of the former one.

Reply:
okay, can refine it a bit. however, it is easier to enforce an unsubscribe in all cases using linting rules

Reply:
Oh don’t get me wrong, the statement itself is fine. I guess to give the full picture gives the reader a better impression and he might better imagine why one hast to unsubscribe in the first place